### PR TITLE
Fix rendering of add-capacity modal for smaller screen

### PR DESCRIPTION
**Pull request by SanjalKatiyar**
URL: https://issues.redhat.com/browse/RHSTOR-1886

**Before:**
Screenshot from 2021-06-08 15-32-57

**After:**
Screenshot from 2021-06-08 15-34-22

### CODE REVIEW

**Comment by GowthamShanmugam** [comment]
I think this is causing this issue:  https://bugzilla.redhat.com/show_bug.cgi?id=1968690 (storage class dropdown), Try to replace <Form> by <form> and try once: https://github.com/openshift/console/pull/8380#issuecomment-856168824

**Comment by SanjalKatiyar** [comment]
> I think this is causing this issue: https://bugzilla.redhat.com/show_bug.cgi?id=1968690 (storage class dropdown), Try to replace
> 
> by and try once: [#8380 (comment)](https://github.com/openshift/console/pull/8380#issuecomment-856168824)

thanks... this works as well... made changes similar to the comment...
warped the <form> around <div> though as otherwise modal was spanning across entire screen for smaller screen sizes....

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx [hunk 1]
```diff
@@ -168,87 +168,86 @@ export const AddCapacityModal = (props: AddCapacityModalProps) => {
...
-                isDisabled
-                id="request-size"
-                className={classNames('pf-c-form-control', 'ceph-add-capacity__input')}
-                type="number"
-                name="requestSize"
-                value={osdSizeWithoutUnit}
-                aria-label="requestSize"
-                data-test-id="requestSize"
+    <div>
+      <form onSubmit={submit} className="modal-content modal-content--no-inner-scroll">
```
Comment (GowthamShanmugam) [inline frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx:172]: try
<form onSubmit={submit} name="form">
  <div className="modal-content modal-content--no-inner-scroll">
  .......
  </div>
</form>

Comment (SanjalKatiyar) [reply frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx]: ack...

Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx]: is it working now?

Comment (SanjalKatiyar) [reply frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx]: yes... I checked with and without `<div>`...

**Review by GowthamShanmugam — APPROVED** [review]

**Comment by GowthamShanmugam** [comment]
@afreen23 Changes look good to me, Please take a look once

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx [hunk 2]
```diff
@@ -168,87 +168,86 @@ export const AddCapacityModal = (props: AddCapacityModalProps) => {
...
-                isDisabled
-                id="request-size"
-                className={classNames('pf-c-form-control', 'ceph-add-capacity__input')}
-                type="number"
-                name="requestSize"
-                value={osdSizeWithoutUnit}
-                aria-label="requestSize"
-                data-test-id="requestSize"
+    <form onSubmit={submit} name="form">
+      <div className="modal-content modal-content--no-inner-scroll">
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx:172]: why do we need <div> ?
can these classes be on <form> ?

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx [hunk 3]
```diff
@@ -168,87 +168,86 @@ export const AddCapacityModal = (props: AddCapacityModalProps) => {
...
-              <TextInput
-                isDisabled
-                id="request-size"
-                className={classNames('pf-c-form-control', 'ceph-add-capacity__input')}
-                type="number"
-                name="requestSize"
-                value={osdSizeWithoutUnit}
-                aria-label="requestSize"
-                data-test-id="requestSize"
+    <form onSubmit={submit} name="form">
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx:171]: Please add a comment with the Bz link for this change.

**Review by afreen23 — CHANGES_REQUESTED** [review]

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx (hunk 2 above)
Comment (SanjalKatiyar) [reply frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx:175]: without `<div className="modal-content modal-content--no-inner-scroll">` modal is spanning across entire screen size (for small screen)
Screenshot from 2021-06-10 11-09-17

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx (hunk 3 above)
Comment (SanjalKatiyar) [reply frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx:174]: ack...

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx (hunk 2 above)
Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx:175]: make a comment with reason

**Review by afreen23 — CHANGES_REQUESTED** [review]
Make a comment for using div

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx (hunk 2 above)
Comment (SanjalKatiyar) [reply frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx:178]: ack...

File: frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx [hunk 4]
```diff
@@ -168,87 +168,92 @@ export const AddCapacityModal = (props: AddCapacityModalProps) => {
...
-                aria-label="requestSize"
-                data-test-id="requestSize"
+    /** https://bugzilla.redhat.com/show_bug.cgi?id=1968690
+     * The quickest and safest fix for now is to not use <Form> and use a straight <form> instead.
+     */
+    /** Modal is spanning across entire screen (for small screen sizes)
+     * Wrapped components inside a <div> to fix it.
+     */
+    <form onSubmit={submit} name="form">
+      <div className="modal-content modal-content--no-inner-scroll">
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx:178]: Should not this be above div ?
```suggestion
    /** Modal is spanning across entire screen (for small screen sizes)
     * Wrapped components inside a <div> to fix it.
     */
      <div className="modal-content modal-content--no-inner-scroll">
```

**Review by afreen23 — CHANGES_REQUESTED** [review]

**Review by afreen23 — APPROVED** [review]
/lgtm

**Pull request merged**